### DISCUSSION TITLE
Do we need to use the ergo cli to generate code?

### DISCUSSION THREAD

**Discussion by Tochemey** [ergo-services/ergo#180] [Q&A] [answered]
Hello @halturin,

I have seen in some of the documentation about code generation and in this example https://github.com/ergo-services/examples/tree/v300/websocket#used-command that code generation is kind of needed. I am not too fun of generating application code except for IDL. Maybe I am missing something.

**Comment by halturin** [comment]
its `ergo` tool, for boilerplate code generation :)

**Reply by halturin** [reply, marked as answer]
for sure, if you dont like it, you can write your project from scratch manually. its up to you :)

**Comment by halturin** [comment]
use https://github.com/ergo-services/tools/tree/v300 if you want to play with it.

see quick start here https://github.com/ergo-services/ergo/tree/v300?tab=readme-ov-file#quick-start

**Reply by halturin** [reply]
more information about the `ergo` tool https://docs.ergo.services/tools/ergo

**Comment by halturin** [comment]
forgot to mention. since the framework 3.0 is not yet released you should manually pin the "ergo.services/ergo" dependency to the latest commit in v300 branch.
